src/server/controllers/chats/webChats.js

```diff
@@ -0,0 +1,40 @@
+const models = require('../../models/index');
+const { postChat } = require('../../middlewares/index');
+const jwt = require('jsonwebtoken');
+
+require('dotenv').config();
+
+module.exports = {
+  // eslint-disable-next-line max-lines-per-function
+  createAppChat: async (req, res) => {
+    if (req.body.text.length === 0) {
+      return res
+        .status(400)
+        .send({ message: 'Text is required', status: 'failure' });
+    }
+    const token = req.headers.authorization;
+    const userId = jwt.decode(token).id;
+    let channelID;
+    const { incidentId } = req.body;
+    try {
+      const SlackChannel = await models.SlackChannels.findAll({
+        attributes: ['channelId'],
+        where: { incidentId: incidentId },
+      });
+      const { channelId } = SlackChannel[0];
+      channelID = channelId;
+    } catch (err) {
+      return res
+        .status(400)
+        .send({ message: 'incident not found', status: 'failure' });
+    }
+    const chat = await models.SlackWireEvents.create({
+      text: req.body.text,
+      type: 'web_message',
+      userId: userId,
+      channelId: channelID,
+    });
+    postChat.postChat(chat.text);
```

Comment:
If this request to slack fails, it will mess up the entire flow. I think it should be wrapped in a try-catch as well, then the postChat function can throw an error when there's an error connecting to slack.